api/desecapi/management/commands/sync-to-pdns.py

```diff
@@ -0,0 +1,92 @@
+from django.core.management import BaseCommand, CommandError
+from django.db import transaction
+
+from desecapi import pdns
+from desecapi.exceptions import PDNSException
+from desecapi.models import Domain, RRset, RR
+from desecapi.pdns_change_tracker import PDNSChangeTracker
+
+
+class Command(BaseCommand):
+    help = 'Sync RRsets from local API database to pdns.'
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain-name', nargs='*',
+                            help='Domain name to sync. If omitted, will import all API domains.')
+
+    def handle(self, *args, **options):
+        domains = Domain.objects.all()
+
+        if options['domain-name']:
+            domains = domains.filter(name__in=options['domain-name'])
+            domain_names = domains.values_list('name', flat=True)
+
+            for domain_name in options['domain-name']:
+                if domain_name not in domain_names:
+                    raise CommandError('{} is not a known domain'.format(domain_name))
+
+        for domain in domains:
+            self.stdout.write('%s ...' % domain.name, ending='')
+            try:
+                self._sync_domain(domain)
+                self.stdout.write(' synced')
+            except Exception as e:
+                self.stdout.write(' failed')
+                msg = 'Error while processing {}: {}'.format(domain.name, e)
+                raise CommandError(msg)
+
+    @staticmethod
+    @transaction.atomic
+    def _sync_domain(domain):
```

Comment:
We could; I can think of two ways:
- Extract the relevant code from the change tracker (move it somewhere to `pdns.py`)
- Keep it where it is, but give the pdns change tracker the option to only run `pdns_do()`, but not `api_do()`.

I'd rather do both of these at some later point, in combination with the Merkle tree replication catalog. That will require some refactoring in that area anyways.

Reply:
Actually, that's what the PDNSChange classes are for. The whole `rrsets_meaty`, `rrsets_empty` construction can be replaced with instances of `PDNSChange` classes, for which then in a loop can be `pdns_do` be called. That also saves additional caveat of using "REPLACE" instead of "DELETE" for deletion (#220).

Reply:
fixed! Now it is much much nicer, thank you for the suggestion.